docs/rules/spec-components-invalid-map-name.md

```diff
@@ -0,0 +1,95 @@
+# spec-components-invalid-map-name
+
+Requires that specific objects inside `components` MUST use keys that match the regular expression: `^[a-zA-Z0-9\.\-_]+$`
+
+|OAS|Compatibility|
+|---|---|
+|2.0|❌|
+|3.0|✅|
+|3.1|✅|
+
+```mermaid
+flowchart TD
+
+root ==> Components -->  Examples --> Example
```

Comment:
```suggestion
Root ==> Components -->  NamedExamples
```
I believe we call these types like this?

https://github.com/Redocly/redocly-cli/blob/master/packages/core/src/types/oas3.ts#L360-L372

I wouldn't show beyond where the evaluation is occurring on the various "Named" components.